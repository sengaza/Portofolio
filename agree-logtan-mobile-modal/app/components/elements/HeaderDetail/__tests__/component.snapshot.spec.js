import React from 'react';
import { View } from 'react-native';
import { assertSnapshots, mockSetState } from '../../../../utils/TestUtils/snapshot';
import HeaderList from '../index';

jest.mock('../../BackButton', () => 'BackButton');
jest.mock('../../CloseButton', () => 'CloseButton');
jest.mock('../../SearchButton', () => 'SearchButton');
jest.mock('../../SearchTextInput', () => 'SearchTextInput');

describe('HeaderList snapshot testing', () => {
  const configs = [
    {
      props: {},
      desc: 'renders without data'
    },
    {
      props: { back: true },
      desc: 'renders with back'
    },
    {
      props: { close: true },
      desc: 'renders without data'
    },
    {
      props: { search: true },
      desc: 'renders without data'
    },
    {
      state: {},
      props: {
        leftComponent: <View />,
        rightComponent: <View />,
        centerComponent: <View />
      },
      desc: 'renders without data'
    },
    {
      state: { searchVisible: false },
      props: {},
      desc: 'renders with isSearching true'
    }
  ];
  assertSnapshots(HeaderList, configs);
});

describe('_renderLeft', () => {
  const ComponentWithMockedState = mockSetState(HeaderList);
  const getInstance = props => new ComponentWithMockedState(props);
  beforeEach(() => {
    this.props = {
      navigation: {
        navigate: jest.fn(),
        goBack: jest.fn()
      },
      onChangeTxtKeyword: jest.fn()
    };
  });

  afterEach(() => {
    this.props = null;
    jest.clearAllMocks();
  });

  test('_onPress', () => {
    const instance = getInstance(this.props);
    instance._onPress();
    expect(instance.state.searchVisible).toBeTruthy();
  });

  test('_onCancel', () => {
    const instance = getInstance(this.props);
    instance._onPress();
    expect(instance.state.searchVisible).toBeTruthy();
    instance._onCancel();
  });

  test('_renderHeaderSearch', () => {
    const instance = getInstance(this.props);
    instance._onPress();
    instance._renderHeaderSearch();
  });
});
