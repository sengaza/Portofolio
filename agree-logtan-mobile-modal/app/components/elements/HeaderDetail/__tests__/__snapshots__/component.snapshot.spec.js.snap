// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderList snapshot testing renders with back 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#1ea54f",
        "flexDirection": "row",
        "height": 70,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <BackButton
        white={true}
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 4,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View />
    </View>
  </View>
</View>
`;

exports[`HeaderList snapshot testing renders with isSearching true 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#1ea54f",
        "flexDirection": "row",
        "height": 70,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View />
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 4,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View />
    </View>
  </View>
</View>
`;

exports[`HeaderList snapshot testing renders without data 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#1ea54f",
        "flexDirection": "row",
        "height": 70,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View />
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 4,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View />
    </View>
  </View>
</View>
`;

exports[`HeaderList snapshot testing renders without data 2`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#1ea54f",
        "flexDirection": "row",
        "height": 70,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <CloseButton
        white={true}
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 4,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View />
    </View>
  </View>
</View>
`;

exports[`HeaderList snapshot testing renders without data 3`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#1ea54f",
        "flexDirection": "row",
        "height": 70,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View />
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 4,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <SearchButton
        onPress={[Function]}
        visible={false}
      />
    </View>
  </View>
</View>
`;

exports[`HeaderList snapshot testing renders without data 4`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "#1ea54f",
        "flexDirection": "row",
        "height": 70,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View />
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 4,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View />
    </View>
  </View>
</View>
`;
